src/main/java/com/watch/shop/app/service/WatchServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.watch.shop.app.service;
+
+import com.watch.shop.app.model.Watch;
+import com.watch.shop.app.model.WatchStore;
+
+import java.util.Comparator;
+import java.util.List;
+
+import static java.util.Objects.isNull;
+
+public class WatchServiceImpl implements WatchService {
+
+    private final WatchStore watchStore = WatchStore.getInstance();
+
+    public List<Watch> getWatches() {
+        return watchStore.getWatches();
+    }
+
+    public List<Watch> sortWatchesByParam(String param) {
+        if (isNull(param) || param.isBlank()) throw new IllegalArgumentException("Param cannot be empty");
+
+        List<Watch> watches = watchStore.getWatches();
+        if ("price".equals(param)) {
+            watches.sort(Comparator.comparing(Watch::getPrice));
+        } else if ("color".equalsIgnoreCase(param)) {
+            watches.sort(Comparator.comparing(Watch::getColor));
+        } else if ("arrival date".equalsIgnoreCase(param)) {
+            watches.sort(Comparator.comparing(Watch::getArrivalDate));
```

Comment:
I guess you could remove duplication of watches.sort by doing next:

Comparator<Watch> comparator = resolveComparator(param);
watches.sort(comparator);

And in private helper method just use switch or if else to resove comparator